client_new/src/types/client.rs

```diff
@@ -107,19 +109,95 @@ impl Client {
     /// # Example
     /// ```
     /// ```
-    pub async fn record_exists(&mut self, location: Location) -> Result<bool, ClientError> {
+    pub async fn record_exists(&self, location: Location) -> Result<bool, ClientError> {
         let (vault_id, record_id) = location.resolve();
-        let db = self.db.try_read().map_err(|_| ClientError::LockAcquireFailed)?;
+        let db = self.db.try_read()?;
         let contains_record = db.contains_record(vault_id, record_id);
         Ok(contains_record)
     }
 
+    /// Synchronize two vaults of the client so that records are copied from `source` to `target`.
+    /// If `select_records` is `Some` only the specified records are copied, else a full sync
+    /// is performed. If a record already exists at the target, the [`MergePolicy`] applies.
+    ///
+    /// # Example
+    /// ```
+    /// ```
+    pub fn sync_vaults(
+        &self,
+        source: VaultId,
+        target: VaultId,
```

Comment:
It looks like this is a public function, so shouldn't this take `Vec<u8>` vault paths instead of the (basically internal) `VaultId` type?

Reply:
Good catch, yes.

Reply:
Perhaps `VaultId` and other id types should be made private, so the compiler would flag things like these, just a thought.